src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -285,17 +285,45 @@ abstract class AbstractServerTask extends AbstractTask {
 
         // envProjectProps and serverEnvFile take precedence over server.env from configDirectory
         File envFile = new File(serverDirectory, "server.env")
+        Properties serverEnvProps = convertServerEnvToProperties(envFile);
```

Comment:
You need code at line 240 to check for appendingServerEnv. That code is copying over the server.env located in the default configuration directory (src/main/liberty/config) and it will overwrite any server.env file already in the target server directory.